pkg/operator/ceph/csi/config.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2024 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"context"
+
+	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
+
+	csiopv1a1 "github.com/ceph/ceph-csi-operator/api/v1alpha1"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func CreateUpdateRadosNamespace(ctx context.Context, c client.Client, clusterInfo *cephclient.ClusterInfo, cephBlockPoolRadosNamespacedName types.NamespacedName, clusterID, clusterName string) error {
+
+	logger.Info("creating ceph-csi clientProfile cr for rados namespace")
+
+	csiOpConfig := &csiopv1a1.Config{}
+	csiOpConfig.Name = clusterID
+	csiOpConfig.Namespace = cephBlockPoolRadosNamespacedName.Namespace
+	csiOpConfig.Spec = csiopv1a1.ConfigSpec{
+		CephClusterRef: v1.LocalObjectReference{
+			Name: clusterName,
+		},
+		Rbd: &csiopv1a1.RbdConfigSpec{
+			RadosNamespace: cephBlockPoolRadosNamespacedName.Name,
+		},
+	}
+
+	err := clusterInfo.OwnerInfo.SetOwnerReference(csiOpConfig)
+	if err != nil {
+		return errors.Wrapf(err, "failed to set owner reference on config CR %q", csiOpConfig.Name)
+	}
+
+	err = c.Create(ctx, csiOpConfig)
+	if err != nil && !kerrors.IsAlreadyExists(err) {
+		return errors.Wrapf(err, "failed to create ceph-csi clientProfile cr for Rbd %s", csiOpConfig.Name)
+	}
+
+	return nil
+}
+
+func CreateUpdateSubvolumeGroup(ctx context.Context, c client.Client, clusterInfo *cephclient.ClusterInfo, cephFilesystemNamespacedName types.NamespacedName, clusterID, clusterName string) error {
```

Comment:
his name is a bit misleading IMO, we are not creating SubvolumeGroup here, but creating the clientProfile CR that has reference to SubvolumeGroup name.

Reply:
agreed, udpated